### PR TITLE
Replace create sockaddr

### PR DESCRIPTION
**Pull request by askawu**
This PR replaces net pkt parsing in net_context.c by net_pkt_get_src_addr() and net_pkt_get_dst_addr().

- The addrlen parameter of net pkt get address function is still there. It might be superfluous but it's nothing wrong to check the size of sockaddr as it's the generic structure for sockaddr_in and sockaddr_in6.
- In tcp_syn_rcvd(), the addrlen of accept_cb will always be the size of sockaddr rather than size of sockaddr_in or sockaddr_in6.

### CODE REVIEW

**Comment by nashif** [comment]
recheck

File: subsys/net/ip/net_context.c [hunk 1]
```diff
@@ -211,6 +177,7 @@ static int tcp_backlog_find(struct net_pkt *pkt, int *empty_slot)
 static int tcp_backlog_syn(struct net_pkt *pkt, struct net_context *context)
 {
 	int empty_slot = -1;
+	int rv;
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: All the return values use the "ret" as variable name in this file, so lets be consistent about it here and change rv to ret.

File: subsys/net/ip/net_context.c [hunk 2]
```diff
@@ -1292,77 +1275,28 @@ NET_CONN_CB(tcp_synack_received)
 	 * If we receive SYN, we send SYN-ACK and go to SYN_RCVD state.
 	 */
 	if (NET_TCP_FLAGS(tcp_hdr) == (NET_TCP_SYN | NET_TCP_ACK)) {
-		struct sockaddr *laddr;
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: Could we keep the addresses inside this block as there does not seem to be a reason why they would need to be outside?

File: subsys/net/ip/net_context.c [hunk 3]
```diff
@@ -1843,7 +1735,7 @@ NET_CONN_CB(tcp_syn_rcvd)
 
 		context->tcp->accept_cb(new_context,
 					&new_context->remote,
-					addrlen,
+					sizeof(new_context->remote),
```
Comment (jukkar) [inline subsys/net/ip/net_context.c]: I would rather see the actual address length here.

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by askawu** [comment]
Update:
- Follow the naming convention of the return value.
- Keep the remote/local sockaddr inside the necessary block.
- Pass the actual address length to acept callback.

**Review by jukkar — COMMENTED** [review]
The actual functionality looks ok. Could you create some simple tests for example in tests/net/utils/src/main.c that test this functionality. This simple routine can easily tested so we should done it.

**Comment by askawu** [comment]
Ok. I will add some test cases.

**Comment by askawu** [comment]
@pfalcon and @jukkar , does it make sense to you to add test cases of net_pkt_get_src_addr() and net_pkt_get_dst_addr() in tests/net/net_pkt/src/main.c?

**Comment by pfalcon** [comment]
@askawu , based on the functions prefix, yeah, that is suitable location. I'd say it grows too big, but splitting it is apparently a separate matter.

**Comment by jukkar** [comment]
According to function prefix the tests would go to tests/net/net_pkt. But if you look the tests there they are more related to actual net_pkt and net_buf handling, and the tests you need to create are related to IP address management. I suggested a directory in tests/net/utils which has already some IP address parsing tests which are kind of similar what you have here. There is also tests/net/ip-addr directory that could have these tests too.
The actual directory is not that important, the tests are.

**Comment by askawu** [comment]
Update:
- Rebase on the master branch
- Add tests in tests/net/utils/src/main.c.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by pfalcon — APPROVED** [review]

**Pull request merged**